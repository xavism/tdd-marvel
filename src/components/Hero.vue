<template>
    <h1>Hero</h1>
</template>

<script>
export default {
    name: 'Hero'
}
</script>
