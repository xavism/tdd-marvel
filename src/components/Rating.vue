<template>
    <h1>Rating</h1>
</template>

<script>
export default {
    name: 'Rating'
}
</script>
