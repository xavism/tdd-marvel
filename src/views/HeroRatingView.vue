<template>
    <div>
        <Hero/>
        <Rating/>
    </div>
</template>

<script>
import Hero from '@/components/Hero'
import Rating from '@/components/Rating'

export default {
    name: 'HeroRatingView',
    components: {
        Hero,
        Rating
    }
}
</script>
