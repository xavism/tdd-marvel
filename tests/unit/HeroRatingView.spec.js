import { shallowMount } from '@vue/test-utils'
import HeroRatingView from '@/views/HeroRatingView'
import Rating from '@/components/Rating'
import Hero from '@/components/Hero'

describe('HeroRatingView', () => {
    const build = () => {
        const wrapper = shallowMount(HeroRatingView);
    
        return {
          wrapper,
          rating: () => wrapper.find(Rating),
          hero: () => wrapper.find(Hero),
        };
      };

    it('renders the component', () => {
        // arrange
        const { wrapper } = build();
        // assert
        expect(wrapper.html()).toMatchSnapshot();
    });
    
    it('renders the childs', () => {
        // arrange
        const { rating, hero } = build();
    
        // assert
        expect(rating().exists()).toBe(true);
        expect(hero().exists()).toBe(true);
    });
})