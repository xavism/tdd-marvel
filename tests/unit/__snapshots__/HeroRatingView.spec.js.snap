// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroRatingView renders the component 1`] = `
<div>
  <hero-stub></hero-stub>
  <rating-stub></rating-stub>
</div>
`;
